### PR TITLE
Update documentation to include changes in aws jslib v0.9.0

### PR DESCRIPTION
**Pull request by oleiade**
This Pull Request adapts the aws jslib documentation section to fit the changes introduced by [version 0.9.0](https://github.com/grafana/k6-jslib-aws/releases/tag/v0.9.0).

This big release makes almost all the APIs facing the user asynchronous and compatible with the `async/await` syntax. Previous releases exposing synchronous codes will stay out there, untouched. Still, as the documentation website is not versioned, only the documentation for the async version will be available once this PR is merged.

This is a massive PR, I apologize for that, but I still look forward for you folks spotting all the tiny mistakes and "I forgot" occurrences probably scattered across the PR 🙇🏻

### CODE REVIEW

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/00 sign().md
Comment (immavalls) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/SignatureV4/00 sign().md:48]:
```suggestion
import http from 'k6/http'
```

**Review by immavalls — APPROVED** [review]
LGTM, lots of work to change all the examples 💪🏼. I tested most examples in the main pages (AwsConfig, S3Client, KMSClient, SecretsManagerClient, and SignatureV4), and they work very well. 
Unrelated, are we still missing the [Kinesis](https://github.com/grafana/k6-jslib-aws/blob/main/examples/kinesis.js) docs?

File: src/components/shared/blocking-aws-blockquote/blocking-aws-blockquote.view.js
Comment (oleiade) [reply src/components/shared/blocking-aws-blockquote/blocking-aws-blockquote.view.js:9]: Good catch indeed! Before we applied it to everything, but now with operations being asynchronous, the "cost" of operations probably impacts only a subset of operations in practice: such as S3.

I'll make a pass on our APIs, and try to find out which modules would deserve that warning 👍🏻 I will also rephrase accordingly.

Comment (oleiade) [reply src/components/shared/blocking-aws-blockquote/blocking-aws-blockquote.view.js]: I've rephrased the block quote to not be specific to S3 anymore, and kept it only on the S3, KMS, and Signature pages 👍🏻

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 S3Client.md
Comment (heitortsergent) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 S3Client.md:14]: Nice editing here 👍 .

A small suggestion, and I'll leave it up to you, you could shorten the list of actions and say something like: 

"With it, you can do several operations such as list buckets, list objects in a bucket, or download objects from a bucket. For a full list of supported operations, see [Methods](#methods)."

That way we don't have to add to this paragraph if we add new methods here (although I don't know if this changes much?)

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SecretsManagerClient.md
Comment (heitortsergent) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SecretsManagerClient.md:10]: Similar feedback as the previous one on having all the methods listed here.

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SignatureV4.md
Comment (heitortsergent) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SignatureV4.md:8]: Does it make sense to have this quote here if the example doesn't include any updates to use async/await? (maybe a silly question 🫥)

Comment (heitortsergent) [reply src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SignatureV4.md:8]: Oh wait, we're removing this quote from the services that now have async/await. That makes sense. 😅

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/S3Client/00 abortMultipartUpload(bucketName, objectKey, uploadId).md
Comment (heitortsergent) [inline src/data/markdown/docs/20 jslib/01 jslib/01 aws/S3Client/00 abortMultipartUpload(bucketName, objectKey, uploadId).md:9]: Nice catch 🙇 .

**Review by heitortsergent — APPROVED** [review]
I left a few comments, mostly suggestions 😊. Thanks for updating all those examples, that must have taken quite some time!

File: src/components/shared/blocking-aws-blockquote/blocking-aws-blockquote.view.js
Comment (oleiade) [reply src/components/shared/blocking-aws-blockquote/blocking-aws-blockquote.view.js:17]: It is indeed 🤝 

As a non-native speaker, I love to use "thus" and "therefore" because it gives me a sense of speaking proper English, so that's good feedback for me to 👂🏻 that they're probably overkill, thanks a lot 🙇🏻

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 S3Client.md
Comment (oleiade) [reply src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 S3Client.md:14]: You're absolutely right, I was actually looking for a way to make it more generic indeed (and not needing to much updates in the future either). I'll rephrase 👍🏻

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SecretsManagerClient.md
Comment (oleiade) [reply src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SecretsManagerClient.md:10]: I'll rephrase too 👍🏻

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SignatureV4.md
Comment (oleiade) [reply src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SignatureV4.md:8]: I've left it here because although it doesn't use async await (yet), it involves some pretty costly cryptographic operations. I would leave it as is in the context of this specific PR, and would reiterate once either the signature code is async, or we rephrase this block completely 👍🏻

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/S3Client/00 abortMultipartUpload(bucketName, objectKey, uploadId).md
Comment (oleiade) [reply src/data/markdown/docs/20 jslib/01 jslib/01 aws/S3Client/00 abortMultipartUpload(bucketName, objectKey, uploadId).md:9]: 🙇🏻

File: src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SignatureV4.md
Comment (heitortsergent) [reply src/data/markdown/docs/20 jslib/01 jslib/01 aws/00 SignatureV4.md:8]: That makes sense, thanks for the explanation 🙇 .

File: src/data/markdown/docs/05 Examples/01 Examples/02 http-authentication.md
Comment (oleiade) [reply src/data/markdown/docs/05 Examples/01 Examples/02 http-authentication.md:121]: I've opened a [dedicated issue](https://github.com/grafana/k6-jslib-aws/issues/56) to keep track of this, and not block this PR. We have received a new external contribution, and will have to create a new release soon anyway, so I'll bring the change with the next version's docs change 👍🏻

File: src/components/shared/blocking-aws-blockquote/blocking-aws-blockquote.view.js
Comment (heitortsergent) [inline src/components/shared/blocking-aws-blockquote/blocking-aws-blockquote.view.js:10]:
```suggestion
    In some cases, using this library&apos;s operations might impact performance
    and skew your test results. <br />
```

I think prettier is complaining about these two lines here. 😩

**Pull request merged**
